forward/request_sender.go

```diff
@@ -156,6 +159,33 @@ func (s *requestSender) Send() (res []byte, err error) {
 	}
 }
 
+type _bufPool struct {
+	sync.Pool
+	maxCap int64
+}
+
+func (bp *_bufPool) Get() *bytes.Buffer {
+	const allocLimit = 100 * 1024 * 1024 // cap initial allocations at 100MiB
```

Comment:
this buffer pool is complicated by the limiting -- is that really needed? `sync.Pool` is frequently cleared (on every GC), so I'd drop the extra logic unless we have a good reason to need it.

Reply:
well see... let's try the turn around:
- if we ever happen to forward a 500MiB message... do we always and forever want to be pre-allocating half-GiB buffers for every future message? I for one, say no ;-)
- in fact, I think 100MiB may be too generous in such scenario

Reply:
I would make the buffer pool very simple: just have a prefixed size (E.g., 1-4mb) and always return buffers of that size.

If we see performance issues with that, then we can make it smarter